Account for <base> tag when calculating integrity.

rwjblue opened the pull request:
Fixes #12.

rwjblue committed:
Account for <base> tag when calculating integrity.

rwjblue commented:
Interestingly the test file contents of `foo/test.html` was not being compared, and it was incorrect (until adding the `<base>` tag as I did here) for exactly the same reason as #12.

stefanpenner commented:
Lgtm. I assume this was an existing bug, not related to my work this morning.

stefanpenner commented on index.js:
```diff
@@ -55,10 +56,26 @@ function SRIHashAssets(inputNodes, options) {
+
+    if (!relativePath) { return null; }
+
+    return this.inputPaths[0] + relativePath;
```
should we worry about '../../../` basePaths? (ones that escape out of the root)

rwjblue replied:
Hmm. I'm not sure? I haven't really seen a relative base url like that, is it valid?

rwjblue replied:
Ya, just confirmed (via [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base)) that it can be relative or absolute.  

I agree with @jonathanKingston, we can opt out of SRI if `<base>` resolves to a path above the project root.

rwjblue replied:
OK, I added a guard to ensure that we do not use a `<base>` that doesn't start with `/` (which should prevent escaping out).

rwjblue commented:
@stefanpenner - Yes, it was an existing issue (though in 1.1.1 it assumed _everything_ was relative to / so it was somewhat the reverse problem).

jonathanKingston commented:
Personally I would prefer to turn off integrity checking in that instance just for paranoias sake really.
In fact I would argue anything other than `'/'` could do the same which supplies a nice carrot/stick for people to move over to using a nice base tag. (That is however me, I am cruel etc)

jonathanKingston commented:
Also again thank you so much for your hard work comrades.

rwjblue committed:
Add guard to prevent escaping out of inputPath.

With a `<base>` of:

```
<base href="../../">
```

We need to make sure that we do not escape out of our base `inputPath`.
This just adds a simple guard to ensure that the base starts at the
root.
